.github/workflows/push.workflow.yml

```diff
@@ -0,0 +1,236 @@
+name: Main branches push checks
+
+on:
+  push:
+    branches:
+      - master
+      - 2-dev
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: |
+            ~/.npm
+            **/node_modules
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - uses: actions/setup-node@v1
+        with:
+          node-version: "12"
+      - uses: ./.github/actions/es-lint
+
+  unit-tests:
+    name: Unit Tests - Node.js v${{ matrix.node-version }}
+    needs: ['lint']
+    strategy:
+      matrix:
+        node-version: ['12.16', '12.18']
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: |
+            ~/.npm
+            **/node_modules
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - uses: ./.github/actions/unit-tests
+
+  integration-tests-legacy:
+    name: Integration Tests Legacy - ${{ matrix.test_set }} - Node.js v${{ matrix.node-version }}
+    needs: ['unit-tests']
+    strategy:
+      matrix:
+        test_set: ['http', 'websocket', 'mqtt']
+        node-version: ['12.16', '12.18']
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: |
+            ~/.npm
+            **/node_modules
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - uses: ./.github/actions/integration-tests
+        env:
+          KUZZLE_FUNCTIONAL_TESTS: test:functional:legacy:${{ matrix.test_set }}
+        with:
+          docker-compose-test-file: ./.ci/test.yml
+
+  integration-tests-sdk:
```

Comment:
I think this is the good opportunity to rename our functional tests:
 - `features/` should be `features-legacy/`
 - `features-sdk/` should be `features/`

And jobs name should reflect this change with a "Functional Tests Legacy" and "Functional Tests"